File: parser/src/main/java/io/ticofab/androidgpxparser/parser/domain/Author.java
package io.ticofab.androidgpxparser.parser.domain;

public class Author {

    private final String mName;
    private final Email mEmail;
    private final Link mLink;

    private Author(Builder builder) {
        this.mName = builder.mName;
        this.mEmail = builder.mEmail;
        this.mLink = builder.mLink;
    }

    public String getName() {
        return mName;
    }

    public Email getEmail() {
        return mEmail;
    }

    public Link getLink() {
        return mLink;
    }

    public static class Builder {
        private String mName;
        private Email mEmail;
        private Link mLink;

        public Builder setName(String name) {
            mName = name;
            return this;
        }

        public Builder setEmail(Email email) {
            mEmail = email;
            return this;
        }

        public Builder setLink(Link link) {
            mLink = link;
            return this;
        }

        public Author build() {
            return new Author(this);
        }
    }
}

File: parser/src/main/kotlin/io/ticofab/androidgpxparser/parser/domain/Author.kt
package io.ticofab.androidgpxparser.parser.domain

class Author private constructor(builder: Builder) {
    val name = builder.name
    val email = builder.email
    val link = builder.link

    class Builder {
        var name: String? = null
            private set
        var email: Email? = null
            private set
        var link: Link? = null
            private set

        fun setName(name: String?) = apply { this.name = name }

        fun setEmail(email: Email?) = apply { this.email = email }

        fun setLink(link: Link?) = apply { this.link = link }

        fun build() = Author(this)
    }
}